packages/collector/test/tracing/cloud/aws/dynamodb/test.js

```diff
@@ -0,0 +1,226 @@
+                simple: withError === false
+              });
+
+              /**
+               * Table takes some time to be available, even though the callback gives a success message
+               */
+              if (operation === 'createTable') {
+                await checkTableExistence(tableName, true);
+              }
+              if (operation === 'deleteTable') {
+                await checkTableExistence(tableName, false);
+              }
+
+              if (operation !== 'deleteTable') {
+                return verify(appControls, response, apiPath, withError);
+              }
+            });
+          });
+        });
+      });
+    });
+
+    function verify(controls, response, apiPath, withError) {
+      return retry(() => {
+        return agentControls.getSpans().then(spans => verifySpans(controls, spans, apiPath, withError));
+      }, retryTime);
+    }
+
+    function verifySpans(controls, spans, apiPath, withError) {
+      const httpEntry = verifyHttpEntry(spans, apiPath, controls);
+      verifyDynamoDBExit(spans, httpEntry, withError);
+
+      if (!withError) {
+        verifyHttpExit(spans, httpEntry, controls);
+      }
+    }
+
+    function verifyHttpEntry(spans, apiPath, controls) {
+      return expectExactlyOneMatching(spans, [
+        span => expect(span.p).to.not.exist,
+        span => expect(span.k).to.equal(constants.ENTRY),
+        span => expect(span.f.e).to.equal(String(controls.getPid())),
+        span => expect(span.f.h).to.equal('agent-stub-uuid'),
+        span => expect(span.n).to.equal('node.http.server'),
+        span => expect(span.data.http.url).to.equal(apiPath)
+      ]);
+    }
+
+    function verifyHttpExit(spans, parentSpan, controls) {
+      return expectExactlyOneMatching(spans, [
+        span => expect(span.t).to.equal(parentSpan.t),
+        span => expect(span.p).to.equal(parentSpan.s),
+        span => expect(span.k).to.equal(constants.EXIT),
+        span => expect(span.f.e).to.equal(String(controls.getPid())),
+        span => expect(span.f.h).to.equal('agent-stub-uuid'),
+        span => expect(span.n).to.equal('node.http.client')
+      ]);
+    }
+
+    function verifyDynamoDBExit(spans, parent, withError) {
+      return expectExactlyOneMatching(spans, [
+        span => expect(span.n).to.equal('dynamodb'),
+        span => expect(span.k).to.equal(constants.EXIT),
+        span => expect(span.t).to.equal(parent.t),
+        span => expect(span.p).to.equal(parent.s),
+        span => expect(span.f.e).to.equal(String(appControls.getPid())),
+        span => expect(span.f.h).to.equal('agent-stub-uuid'),
+        span => expect(span.error).to.not.exist,
+        span => expect(span.ec).to.equal(withError ? 1 : 0),
+        span => expect(span.async).to.not.exist,
+        span => expect(span.data).to.exist,
+        span => expect(span.data.dynamodb).to.be.an('object'),
+        span => expect(span.data.dynamodb.op).to.exist
```

Comment:
Can we pass the expected op into this method to verify the actual value and not only that the annotation _exists_?

Also, are there more annotations that we could/should be capturing and that we should test for? (`table` for example, I think I have seen that below in the instrumentation code).